biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -648,11 +648,15 @@ void doPlugins() {
 		for (AnalyzerPlugin plugin : getPlugins(AnalyzerPlugin.class)) {
 			try {
 				Processor previous = beginHandleErrors(plugin.toString());
-				boolean reanalyze = plugin.analyzeJar(this);
-				endHandleErrors(previous);
-				if (reanalyze) {
-					classspace.clear();
-					analyzeBundleClasspath();
+				try {
+					boolean reanalyze = plugin.analyzeJar(this);
+					if (reanalyze) {
```

Comment:
This is still a behavior change. Before endHandleErrors was called before reanalyzing. Your change calls it after. I think the finally must be right after the plugin.analyzeJar call to avoid unintended behavioral changes.